WcaOnRails/app/models/competition.rb

```diff
@@ -266,11 +294,37 @@ def registration_must_close_after_it_opens
     end
   end
 
+  def use_wca_as_website=(value)
+    @use_wca_as_website = ActiveRecord::Type::Boolean.new.type_cast_from_database(value)
+  end
+
+  def use_wca_as_website
+    @use_wca_as_website || website == url
+  end
+
+  before_validation :assign_wca_website_url
+  private def assign_wca_website_url
+    if use_wca_as_website && website != url
+      self.website = self.url
+    end
+  end
+
+  after_save :update_website
+  private def update_website
+    if use_wca_as_website && website != url
+      update_column :website, url
```

Comment:
So we talked about this in the abstract, but seeing the concrete code it's a little frightening. I don't understand what this line, or line 298, does.

Conceptually, the natural code for this stuff is something like...

`competition_website = @comp.use_external_site ? @comp.external_website : internal_website_for(@comp)`

... and the question is if that goes in the model, view or controller.